In [1]:
import pandas as pd

In [3]:
results_df = pd.read_csv('1345_customer_results.csv',
                         usecols = ['Origin', 'Origin Created At', 'Labeling State', 'Qualified Reads', 'Correct Label', 
                                    'Majority Label', 'Difficulty', 'Agreement', 'First Choice Answer', 'First Choice Votes', 
                                    'First Choice Weight', 'Second Choice Answer', 'Second Choice Votes', 'Second Choice Weight'])
pd.set_option('display.max_colwidth', None)

results_df

Origin  \
0      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B14_time139137_medoid_vote2.jpg   
1      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B10_time178433_medoid_vote0.jpg   
2      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B13_time175617_medoid_vote0.jpg   
3       https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B19_time62657_medoid_vote0.jpg   
4       https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B12_time28993_medoid_vote4.jpg   
...                                                                                                               ...   
30288   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time69658_medoid_vote2.jpg   
30289  https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S93_time217256_member_vote3.jpg   
30290   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time55376_member_vote4.jpg   
30291  https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time206508_medoid_vote5.jpg   
30292   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time31489_member_vote5.jpg   

             Origin Created At Labeling State  Qualified Reads Correct Label  \
0      Jul 26 2021 21:41:47 PM  Gold Standard                2          'no'   
1      Jul 26 2021 21:41:47 PM  Gold Standard                3          'no'   
2      Jul 26 2021 21:41:47 PM  Gold Standard                2          'no'   
3      Jul 26 2021 21:41:47 PM  Gold Standard                1          'no'   
4      Jul 26 2021 21:41:47 PM    In Progress                7           NaN   
...                        ...            ...              ...           ...   
30288  Jul 26 2021 22:00:15 PM  Gold Standard                2          'no'   
30289  Jul 26 2021 22:00:15 PM  Gold Standard                3          'no'   
30290  Jul 26 2021 22:00:15 PM    In Progress                6           NaN   
30291  Jul 26 2021 22:00:15 PM  Gold Standard                0         'yes'   
30292  Jul 26 2021 22:00:15 PM  Gold Standard                5         'yes'   

      Majority Label  Difficulty  Agreement First Choice Answer  \
0               'no'       0.000      1.000                'no'   
1               'no'       0.000      1.000                'no'   
2               'no'       0.000      1.000                'no'   
3               'no'       0.000      1.000                'no'   
4              'yes'         NaN      0.571               'yes'   
...              ...         ...        ...                 ...   
30288          'yes'       1.000      1.000               'yes'   
30289          'yes'       0.667      0.667               'yes'   
30290          'yes'         NaN      1.000               'yes'   
30291            NaN         NaN        NaN               'yes'   
30292          'yes'       0.200      0.800               'yes'   

       First Choice Votes  First Choice Weight Second Choice Answer  \
0                       2                 1.54                'yes'   
1                       3                 2.34                'yes'   
2                       2                 1.70                'yes'   
3                       1                 0.82                'yes'   
4                       4                 3.28                 'no'   
...                   ...                  ...                  ...   
30288                   2                 1.56                 'no'   
30289                   2                 1.56                 'no'   
30290                   6                 4.78                 'no'   
30291                   0                 0.00                 'no'   
30292                   4                 3.20                 'no'   

       Second Choice Votes  Second Choi

- Removed the extra empty columns that won't serve as anything useful

In [4]:
results_df['Expert Yes Votes'] = results_df['Origin'].str[-5]
results_df['Expert Yes Votes'] = results_df['Expert Yes Votes'].astype(int)

In [5]:
results_df['Expert Yes Votes'].value_counts()

1    5124
2    3627
3    3254
4    3118
5    3066
6    3037
7    3025
8    3016
0    3012
9      14
Name: Expert Yes Votes, dtype: int64

In [6]:
#Drop rows where there are 9 votes:
results_df = results_df[results_df['Expert Yes Votes'] < 9]

In [7]:
results_df['Expert Yes Votes'].value_counts()

1    5124
2    3627
3    3254
4    3118
5    3066
6    3037
7    3025
8    3016
0    3012
Name: Expert Yes Votes, dtype: int64

In [10]:
#Expert majority answers:
results_df.loc[results_df['Expert Yes Votes'] > 4,'Expert Label'] = "'yes'"
results_df.loc[results_df['Expert Yes Votes'] < 4,'Expert Label'] = "'no'"

In [11]:
results_df

Origin  \
0      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B14_time139137_medoid_vote2.jpg   
1      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B10_time178433_medoid_vote0.jpg   
2      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B13_time175617_medoid_vote0.jpg   
3       https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B19_time62657_medoid_vote0.jpg   
4       https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B12_time28993_medoid_vote4.jpg   
...                                                                                                               ...   
30288   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time69658_medoid_vote2.jpg   
30289  https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S93_time217256_member_vote3.jpg   
30290   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time55376_member_vote4.jpg   
30291  https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time206508_medoid_vote5.jpg   
30292   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time31489_member_vote5.jpg   

             Origin Created At Labeling State  Qualified Reads Correct Label  \
0      Jul 26 2021 21:41:47 PM  Gold Standard                2          'no'   
1      Jul 26 2021 21:41:47 PM  Gold Standard                3          'no'   
2      Jul 26 2021 21:41:47 PM  Gold Standard                2          'no'   
3      Jul 26 2021 21:41:47 PM  Gold Standard                1          'no'   
4      Jul 26 2021 21:41:47 PM    In Progress                7           NaN   
...                        ...            ...              ...           ...   
30288  Jul 26 2021 22:00:15 PM  Gold Standard                2          'no'   
30289  Jul 26 2021 22:00:15 PM  Gold Standard                3          'no'   
30290  Jul 26 2021 22:00:15 PM    In Progress                6           NaN   
30291  Jul 26 2021 22:00:15 PM  Gold Standard                0         'yes'   
30292  Jul 26 2021 22:00:15 PM  Gold Standard                5         'yes'   

      Majority Label  Difficulty  Agreement First Choice Answer  \
0               'no'       0.000      1.000                'no'   
1               'no'       0.000      1.000                'no'   
2               'no'       0.000      1.000                'no'   
3               'no'       0.000      1.000                'no'   
4              'yes'         NaN      0.571               'yes'   
...              ...         ...        ...                 ...   
30288          'yes'       1.000      1.000               'yes'   
30289          'yes'       0.667      0.667               'yes'   
30290          'yes'         NaN      1.000               'yes'   
30291            NaN         NaN        NaN               'yes'   
30292          'yes'       0.200      0.800               'yes'   

       First Choice Votes  First Choice Weight Second Choice Answer  \
0                       2                 1.54                'yes'   
1                       3                 2.34                'yes'   
2                       2                 1.70                'yes'   
3                       1                 0.82                'yes'   
4                       4                 3.28                 'no'   
...                   ...                  ...                  ...   
30288                   2                 1.56                 'no'   
30289                   2                 1.56                 'no'   
30290                   6                 4.78                 'no'   
30291                   0                 0.00                 'no'   
30292                   4                 3.20                 'no'   

       Second Choice Votes  Second Choi

# Does the expert majority agree with the crowd majority?

In [17]:
results_df.loc[results_df["Expert Label"] == results_df["Majority Label"],'Expert Label Agrees w/ Crowd Majority?'] = "'agree'"
results_df.loc[results_df["Expert Label"] != results_df["Majority Label"],'Expert Label Agrees w/ Crowd Majority?'] = "'disagree'"

In [18]:
results_df["Expert Label Agrees w/ Crowd Majority?"].value_counts('agree')

'agree'       0.706661
'disagree'    0.293339
Name: Expert Label Agrees w/ Crowd Majority?, dtype: float64

- In 70.6661% of the 30279 counts of data, the expert majority agrees with the crowd majority

# Who Performs Better: Crowd or Experts?

In [20]:
results_df.loc[results_df["Expert Label"] == results_df["Correct Label"],'Expert Label Correct?'] = "'correct'"
results_df.loc[results_df["Expert Label"] != results_df["Correct Label"],'Expert Label Correct?'] = "'incorrect'"

results_df.loc[results_df["Majority Label"] == results_df["Correct Label"],'Majority Label Correct?'] = "'correct'"
results_df.loc[results_df["Majority Label"] != results_df["Correct Label"],'Majority Label Correct?'] = "'incorrect'"


In [21]:
results_df

Origin  \
0      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B14_time139137_medoid_vote2.jpg   
1      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B10_time178433_medoid_vote0.jpg   
2      https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B13_time175617_medoid_vote0.jpg   
3       https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B19_time62657_medoid_vote0.jpg   
4       https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/B12_time28993_medoid_vote4.jpg   
...                                                                                                               ...   
30288   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time69658_medoid_vote2.jpg   
30289  https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S93_time217256_member_vote3.jpg   
30290   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time55376_member_vote4.jpg   
30291  https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time206508_medoid_vote5.jpg   
30292   https://centaur-customer-uploads.s3.us-east-1.amazonaws.com/mgh-eeg/210721/jpg/S98_time31489_member_vote5.jpg   

             Origin Created At Labeling State  Qualified Reads Correct Label  \
0      Jul 26 2021 21:41:47 PM  Gold Standard                2          'no'   
1      Jul 26 2021 21:41:47 PM  Gold Standard                3          'no'   
2      Jul 26 2021 21:41:47 PM  Gold Standard                2          'no'   
3      Jul 26 2021 21:41:47 PM  Gold Standard                1          'no'   
4      Jul 26 2021 21:41:47 PM    In Progress                7           NaN   
...                        ...            ...              ...           ...   
30288  Jul 26 2021 22:00:15 PM  Gold Standard                2          'no'   
30289  Jul 26 2021 22:00:15 PM  Gold Standard                3          'no'   
30290  Jul 26 2021 22:00:15 PM    In Progress                6           NaN   
30291  Jul 26 2021 22:00:15 PM  Gold Standard                0         'yes'   
30292  Jul 26 2021 22:00:15 PM  Gold Standard                5         'yes'   

      Majority Label  Difficulty  Agreement First Choice Answer  \
0               'no'       0.000      1.000                'no'   
1               'no'       0.000      1.000                'no'   
2               'no'       0.000      1.000                'no'   
3               'no'       0.000      1.000                'no'   
4              'yes'         NaN      0.571               'yes'   
...              ...         ...        ...                 ...   
30288          'yes'       1.000      1.000               'yes'   
30289          'yes'       0.667      0.667               'yes'   
30290          'yes'         NaN      1.000               'yes'   
30291            NaN         NaN        NaN               'yes'   
30292          'yes'       0.200      0.800               'yes'   

       First Choice Votes  First Choice Weight Second Choice Answer  \
0                       2                 1.54                'yes'   
1                       3                 2.34                'yes'   
2                       2                 1.70                'yes'   
3                       1                 0.82                'yes'   
4                       4                 3.28                 'no'   
...                   ...                  ...                  ...   
30288                   2                 1.56                 'no'   
30289                   2                 1.56                 'no'   
30290                   6                 4.78                 'no'   
30291                   0                 0.00                 'no'   
30292                   4                 3.20                 'no'   

       Second Choice Votes  Second Choi

In [22]:
results_df['Expert Label Correct?'].value_counts()

'correct'      27017
'incorrect'     3262
Name: Expert Label Correct?, dtype: int64

In [23]:
results_df['Majority Label Correct?'].value_counts()

'correct'      21617
'incorrect'     8662
Name: Majority Label Correct?, dtype: int64

In [24]:
#Expert Correct Percentage:
results_df['Expert Label Correct?'].value_counts('correct')

'correct'      0.892269
'incorrect'    0.107731
Name: Expert Label Correct?, dtype: float64

In [25]:
#Majority Correct Percentage:
results_df['Majority Label Correct?'].value_counts('correct')

'correct'      0.713927
'incorrect'    0.286073
Name: Majority Label Correct?, dtype: float64

## Experts got 89.2269% of the labeling correct.
## Majority got 71.3927% of the labeling correct.